core/test-runner.ts

```diff
@@ -18,6 +19,16 @@ export class TestRunner {
    private _testResults = new TestSetResults();
    private _currentTestResults: TestResults;
    private _currentTestFixtureResults: TestFixtureResults;
+   private _output: TestOutput;
+
+   constructor (output?: TestOutput) {
+       // If we were given a TestOutput, use it, otherwise make one
+       if (output !== undefined) {
```

Comment:
I thought it would be better to keep it within the `TestRunner` object because it means that people using the API (eventually) don't need to worry about having factories etc.